### PR TITLE
ettercap: fix build for Linux and ARM

### PR DESCRIPTION
**Pull request by cho-m**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

https://github.com/Homebrew/homebrew-core/runs/3063394449?check_suite_focus=true
```
-- Found Curses: /home/linuxbrew/.linuxbrew/lib/libncurses.so  
-- Some or all of the gtk libraries were not found. (missing: GTK3_GLIBCONFIG_INCLUDE_DIR) 
-- Some or all of the gtk libraries were not found. (missing: GTK3_GLIBCONFIG_INCLUDE_DIR) 
CMake Error at cmake/Modules/EttercapLibCheck.cmake:62 (message):
  You choose to build against GTK3.Please install it, or build against GTK1
Call Stack (most recent call first):
  CMakeLists.txt:106 (include)
```

---

Might as well try ARM too. The fix for non-/usr/local/ Homebrew directory could have impacted build there.

### CODE REVIEW

File: Formula/ettercap.rb [hunk 1]
```diff
@@ -22,11 +22,22 @@ class Ettercap < Formula
...
   def install
     # Work around a CMake bug affecting harfbuzz headers and pango
     # https://gitlab.kitware.com/cmake/cmake/issues/19531
     ENV.append_to_cflags "-I#{Formula["harfbuzz"].opt_include}/harfbuzz"
 
+    # Fix build error on wdg_file.c: fatal error: menu.h: No such file or directory
+    on_linux { ENV.append_to_cflags "-I#{Formula["ncurses"].opt_include}/ncursesw" }
+
+    # Fix missing: GTK3_GLIBCONFIG_INCLUDE_DIR
+    inreplace "cmake/Modules/FindGTK3.cmake", "/usr/local/lib", HOMEBREW_PREFIX/"lib"
```
Comment (carlocab) [inline Formula/ettercap.rb:39]: Can we pass this as a `-D` flag to CMake instead of an `inreplace`?

**Review by carlocab — COMMENTED** [review]
Let's try an ARM build too; your fixes might help there. Though we might have to remove some of the `on_linux` blocks.

**Comment by cho-m** [comment]
ARM build did succeed. So, we can serially bottle both ARM and Linux after merge.

**Comment by carlocab** [comment]
Ah, right. Missed the `CI-force-arm` flag.

File: Formula/ettercap.rb (hunk 1 above)
Comment (cho-m) [reply Formula/ettercap.rb:39]: Wasn't familiar enough with the CMake logic to see if there was a way.

The inreplace is modifying the function find behavior at
https://github.com/Ettercap/ettercap/blob/v0.8.3.1/cmake/Modules/FindGTK3.cmake#L160-L187

On Linux, the first issue seen was running `_gtk3_find_include_dir(GTK3_GLIBCONFIG_INCLUDE_DIR glibconfig.h)`.

Comment (cho-m) [reply Formula/ettercap.rb:39]: Maybe using a default CMake path could help. Just need to check if any work.
I might give `CMAKE_FIND_ROOT_PATH` a try

Comment (carlocab) [reply Formula/ettercap.rb:39]: We could probably send a patch to the upstream project adding `/opt/homebrew/lib` to their search paths.

Or maybe even better (for users building from source in a non-default prefix), some CMake code that queries `brew` for `brew --prefix`. Not sure they'll be keen on carrying this code around, in which case we'll want to keep the `inreplace`, but it's worth a shot.

Comment (cho-m) [reply Formula/ettercap.rb:39]: I think there are some CMake variables we can use. Still going through documentation in https://cmake.org/cmake/help/latest/command/find_path.html but maybe one of:
- `GTK3_ROOT=#{Formula["gtk+3"]}.opt_prefix`
- `CMAKE_PREFIX_PATH=#{HOMEBREW_PREFIX}`
- `CMAKE_INCLUDE_PATH=#{HOMEBREW_PREFIX}/include`

Comment (carlocab) [reply Formula/ettercap.rb:39]: `brew` already sets `CMAKE_PREFIX_PATH` in superenv, so it's probably something else.

Comment (cho-m) [reply Formula/ettercap.rb:39]: Passing in the path for the problematic variable does work and will probably have the lowest impact:
```
-DGTK3_GLIBCONFIG_INCLUDE_DIR=#{Formula["glib"].opt_lib}/glib-2.0/include
```

Comment (carlocab) [reply Formula/ettercap.rb:39]: Sounds fine to me.

**Review by SMillerDev — APPROVED** [review]

**Comment by carlocab** [comment]
ARM bottle dispatched in https://github.com/Homebrew/homebrew-core/actions/runs/1055330150. We can build this for Linux when that's done.

**Pull request merged**